injectproxy/routes.go

```diff
@@ -41,6 +41,8 @@ type routes struct {
 	handler    http.Handler
 	label      string
```

Comment:
For clarity, I would remove the `label` field and rely only on headerName/queryName. Accordingly the NewRoutes() signature would need to be updated to

```
NewRoutes(upstream *url.URL, opts ...Option)
````

And an error returned if none of headerName, queryName or labelValue are set.

Reply:
Not sure, label is still referenced on some places